support/jwt/src/main/kotlin/org/doorip/support/jwt/JwtProvider.kt

```
@@ -0,0 +1,87 @@
+    fun validateAndGetClaim(token: String, key: SignatureKey, claimName: String): String? {
+        val payload = parsePayload(token, key) ?: return null
+        return payload.get(claimName, String::class.java)
+    }
+
+    fun parseHeader(token: String): Map<String, String>? {
+        try {
+            val header = token.split(".")[0]
+            val decoder = Base64.getUrlDecoder()
+            val decoded = String(decoder.decode(header))
+            return objectMapper.readValue(decoded, typeReference)
+        } catch (e: DatabindException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)